src/templates/writeup.js

```diff
@@ -0,0 +1,54 @@
+/** @jsx jsx */
+import { jsx } from 'theme-ui'
+import { graphql } from 'gatsby'
+import { Global } from '@emotion/core'
+
+import unified from 'unified'
+import rehypeReact from '../utils/renderer'
+import rehypeSlug from 'rehype-slug'
+
+import Layout from '../components/layout'
+import Hero from '../components/hero'
+import Container from '../components/container'
+
+const processor = unified()
+  .use(rehypeSlug)
```

Comment:
Maybe using something like ``remark-toc`` or ``mdast-util-toc`` for TOC should help with finding problems in long writeups